lib/galaxy/tools/toolbox/watcher.py

```diff
@@ -151,7 +153,7 @@ def __init__(self, toolbox, observer_class):
         self.start()
 
     def start(self):
-        self.observer.start()
+        register_postfork_function(self.observer.start)
```

Comment:
I agree it shouldn't be uWSGI specific, which is why I put it in `galaxy.util.postfork` instead of something more uWSGI-specific. This is potentially a problem for any server that uses the traditional load-once-fork-many multiprocess model, not just uWSGI.

I think a name like `handle_long_running_function` would be a bit of a misnomer here since this is not really what it does. There could be a better name that indicates that this is a function that will be run "now or later depending on your multiprocess model".